# Complex Numbers – In Depth

---

##  1. What Are Complex Numbers?

A <b>complex number</b> is a number that combines a <b>real</b> part and an <b>imaginary</b> part.

<h4>General Form:</h4>

$$
z = a + bi
$$

<ul>
  <li><b>a</b> → Real part: \( \text{Re}(z) = a \)</li>
  <li><b>b</b> → Imaginary part: \( \text{Im}(z) = b \)</li>
  <li><b>i</b> → Imaginary unit, where \( i^2 = -1 \)</li>
</ul>

<h4>Example:</h4>

$$
z = 3 + 4i \quad \text{(Real part: 3, Imaginary part: 4)}
$$

---

##  2. Understanding the Imaginary Unit \( i \)

The imaginary unit \( i \) is defined such that:

$$
i^2 = -1
$$

So we get:

$$
i^0 = 1,\quad i^1 = i,\quad i^2 = -1,\quad i^3 = -i,\quad i^4 = 1, \dots
$$

---

##  3. Complex Plane (Argand Diagram)

A complex number can be represented as a point or vector on a 2D plane:

<ul>
  <li>X-axis: Real part</li>
  <li>Y-axis: Imaginary part</li>
</ul>

<img src="../images/ArgandDiagram.png" width="800"/>

---

##  4. Operations with Complex Numbers

Let \( z_1 = a + bi \), \( z_2 = c + di \)

### ➕ Addition:
$$
z_1 + z_2 = (a + c) + (b + d)i
$$

### ➖ Subtraction:
$$
z_1 - z_2 = (a - c) + (b - d)i
$$

### ✖ Multiplication:
$$
z_1 z_2 = (ac - bd) + (ad + bc)i
$$

### ➗ Division:
$$
\frac{z_1}{z_2} = \frac{a + bi}{c + di} = \frac{(a + bi)(c - di)}{c^2 + d^2}
$$

Multiply numerator and denominator by the <b>conjugate</b> of the denominator.

---

##  5. Complex Conjugate

The <b>conjugate</b> of \( z = a + bi \) is:

$$
\bar{z} = a - bi
$$

<b>Properties:</b>

- \( z \cdot \bar{z} = a^2 + b^2 = |z|^2 \)
- Useful in division and simplifying expressions

---

##  6. Modulus (Magnitude) of a Complex Number

The <b>modulus</b> or absolute value is the distance from the origin in the complex plane:

$$
|z| = \sqrt{a^2 + b^2}
$$

---

##  7. Argument (Phase Angle)

The <b>argument</b> is the angle (in radians) between the positive real axis and the line connecting the origin to the point:

$$
\arg(z) = \theta = \tan^{-1} \left( \frac{b}{a} \right)
$$

---

##  8. Polar and Exponential Form

A complex number can also be represented as:

### Polar Form:

$$
z = r(\cos \theta + i \sin \theta)
$$

Where:
- \( r = |z| \)
- \( \theta = \arg(z) \)

### Euler's Formula:

$$
z = re^{i\theta}
$$

---

##  9. De Moivre’s Theorem

For integer \( n \), and \( z = re^{i\theta} \):

$$
z^n = r^n e^{in\theta} = r^n (\cos(n\theta) + i \sin(n\theta))
$$

---

##  10. Roots of Complex Numbers

To find \( n \)-th roots of \( z = re^{i\theta} \):

$$
\sqrt[n]{z} = r^{1/n} e^{i(\theta + 2k\pi)/n},\quad k = 0, 1, ..., n-1
$$

There are \( n \) distinct roots, equally spaced around a circle in the complex plane.

---

##  11. Applications

<table>
  <thead>
    <tr><th>Field</th><th>Use Case</th></tr>
  </thead>
  <tbody>
    <tr><td>Electrical Engineering</td><td>AC circuits and phasors</td></tr>
    <tr><td>Quantum Mechanics</td><td>Wavefunctions use complex amplitudes</td></tr>
    <tr><td>Signal Processing</td><td>Fourier Transform, filtering</td></tr>
    <tr><td>Control Systems</td><td>Poles/zeros in transfer functions</td></tr>
    <tr><td>Fractals & Chaos</td><td>Mandelbrot and Julia sets</td></tr>
    <tr><td>2D Rotations</td><td>Use of Euler form for rotation</td></tr>
  </tbody>
</table>

---

##  12. Python Example

```python
import cmath

z1 = 3 + 4j
z2 = 1 - 2j

print("Addition:", z1 + z2)
print("Multiplication:", z1 * z2)
print("Conjugate:", z1.conjugate())
print("Modulus:", abs(z1))
print("Phase:", cmath.phase(z1))
print("Polar Form:", cmath.polar(z1))
print("Rectangular from polar:", cmath.rect(5, cmath.pi/3))


In [1]:
import cmath

z1 = 3 + 4j
z2 = 1 - 2j

print("Addition:", z1 + z2)
print("Multiplication:", z1 * z2)
print("Conjugate:", z1.conjugate())
print("Modulus:", abs(z1))
print("Phase:", cmath.phase(z1))
print("Polar Form:", cmath.polar(z1))
print("Rectangular from polar:", cmath.rect(5, cmath.pi/3))

Addition: (4+2j)
Multiplication: (11-2j)
Conjugate: (3-4j)
Modulus: 5.0
Phase: 0.9272952180016122
Polar Form: (5.0, 0.9272952180016122)
Rectangular from polar: (2.5000000000000004+4.330127018922193j)


<hr>

### Polar Coordinate Representation

<img src="../images/polar_coordiante_representation.png">
<hr>
<img src="../images/polar_coordiante_representation_cos_sine_tang.png">
<hr>
<img src="../images/polar_coordiante_representation_lambda.png">
<hr>
<img src="../images/polar_coordiante_representation_1.png">
<hr>
<img src="../images/polar_coordiante_representation_2.png">

<h2>Understanding Complex Numbers in Polar Coordinates</h2>

<h3>Rectangular (Cartesian) Form</h3>

Any complex number <code>c</code> can be written as:

$$
c = a + ib
$$

<ul>
  <li><strong>a</strong> → real part (x-axis)</li>
  <li><strong>b</strong> → imaginary part (y-axis)</li>
  <li><strong>i</strong> → imaginary unit where \( i^2 = -1 \)</li>
</ul>

---

<h3>Polar Form of a Complex Number</h3>

We can also represent a complex number in terms of <strong>magnitude</strong> and <strong>angle</strong>:

$$
c = |c| \cdot (\cos(\gamma) + i \sin(\gamma))
$$

Where:

<ul>
  <li><code>|c|</code> is the <strong>magnitude</strong> (or modulus) of the complex number</li>
  <li><code>γ</code> is the <strong>angle</strong> (or argument) between the real axis and the vector to the point <code>c</code></li>
</ul>

---

<h3>Conversion Between Forms</h3>

Using the right triangle in the image:

<ul>
  <li><strong>Real part:</strong><br/>
  \( a = |c| \cdot \cos(\gamma) \)
  </li><br/>
  <li><strong>Imaginary part:</strong><br/>
  \( b = |c| \cdot \sin(\gamma) \)
  </li><br/>
  <li><strong>Magnitude (modulus):</strong><br/>
  \( |c| = \sqrt{a^2 + b^2} \)
  </li><br/>
  <li><strong>Angle (argument):</strong><br/>
  \( \gamma = \arctan\left(\frac{b}{a}\right) \)
  </li>
</ul>

---

<h3>Why Polar Form?</h3>

Polar form simplifies complex number operations:

<p><strong>Multiplication:</strong></p>

$$
z_1 \cdot z_2 = |z_1||z_2| \cdot [\cos(\theta_1 + \theta_2) + i \sin(\theta_1 + \theta_2)]
$$

<p><strong>Applications:</strong></p>

<ul>
  <li>Signal processing</li>
  <li>Fourier transforms</li>
  <li>Electrical engineering (phasors)</li>
  <li>Rotations in the complex plane</li>
</ul>

---

<h3>Image Summary</h3>

The diagram shows:

<ul>
  <li><code>c</code> as a point in the complex plane</li>
  <li><code>a</code> and <code>b</code> as Cartesian projections</li>
  <li>A right triangle formed by the vector and axes</li>
  <li>Trigonometric relationships defining cosine and sine</li>
</ul>

---

<h3>Final Takeaway</h3>

Polar coordinates offer an <strong>elegant geometric view</strong> of complex numbers:

$$
c = |c| e^{i\gamma}
$$

This is based on <strong>Euler’s formula</strong>:

$$
e^{i\gamma} = \cos(\gamma) + i\sin(\gamma)
$$

---



<img src="../images/direction_polar.png">
<hr>
<img src="../images/direction_polar_1.png">
<hr>
<img src="../images/direction_polar_2.png">
<hr>
<img src="../images/direction_polar_3.png">
<hr>
<img src="../images/direction_polar_4.png">
<hr>
<img src="../images/direction_polar_5.png">
<hr>


<h3>Special Case: When Imaginary Part is 0 and Real Part is -1</h3>
<img src="../images/eulers_identity.png">
<p>This corresponds to a point on the negative real axis in the complex plane.</p>

<p>Let's use <strong>Euler’s formula</strong>:</p>

$$
e^{i\gamma} = \cos(\gamma) + i \sin(\gamma)
$$

<p>Now, set <code>γ = π</code> (180 degrees):</p>

$$
e^{i\pi} = \cos(\pi) + i \sin(\pi) = -1 + i \cdot 0 = -1
$$

<p>This is the famous identity:</p>

$$
e^{i\pi} + 1 = 0
$$

<p><strong>Interpretation:</strong></p>
<ul>
  <li>Imaginary part <code>i·sin(π) = 0</code></li>
  <li>Real part <code>cos(π) = -1</code></li>
  <li>So, <code>e^{iπ}</code> lands at -1 on the real axis</li>
</ul>

<p>This is a key point on the unit circle, known as one of the most beautiful equations in mathematics because it connects:</p>

<ul>
  <li><code>e</code> — the base of natural logarithms</li>
  <li><code>i</code> — the imaginary unit</li>
  <li><code>π</code> — the circle constant</li>
  <li><code>1</code> — the multiplicative identity</li>
  <li><code>0</code> — the additive identity</li>
</ul>

<h4>Conclusion</h4>

<p>
So when the imaginary part is 0 and the real part is <code>-1</code>, the complex number lies on the left edge of the unit circle and represents the rotation of π radians (180°). This beautifully cancels to zero when added to 1


<hr>
<h2>Additive Identity: <code>0</code></h2>

<h3>Definition:</h3>
<p>The <strong>additive identity</strong> is the number that, when added to any number, leaves the number unchanged.</p>

<h3>Why 0?</h3>
<p>Because:</p>
<ul>
  <li>For any number <code>a</code>, we have:</li>
  <li><code>a + 0 = 0 + a = a</code></li>
</ul>

<h4>Example:</h4>
<ul>
  <li><code>7 + 0 = 7</code></li>
  <li><code>(-5.3) + 0 = -5.3</code></li>
</ul>

<p>So "identity" here means: it doesn’t change the value when added.</p>

<hr>

<h2>Multiplicative Identity: <code>1</code></h2>

<h3>Definition:</h3>
<p>The <strong>multiplicative identity</strong> is the number that, when multiplied by any number, leaves the number unchanged.</p>

<h3>🔸 Why 1?</h3>
<p>Because:</p>
<ul>
  <li>For any number <code>a</code>, we have:</li>
  <li><code>a × 1 = 1 × a = a</code></li>
</ul>

<h4>Example:</h4>
<ul>
  <li><code>4 × 1 = 4</code></li>
  <li><code>(-2.5) × 1 = -2.5</code></li>
</ul>

<p>So this identity doesn't change the value when multiplied.</p>

<hr>
